projects/cardiovascular-patient-app/src/app/app.component.spec.ts

```diff
@@ -1,22 +1,74 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { AppComponent } from './app.component';
 import { AppModule } from './app.module';
+import { MatTabGroupHarness } from '@angular/material/tabs/testing';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatTabsModule } from '@angular/material/tabs';
+
 
 describe('AppComponent', () => {
   let component: AppComponent;
   let fixture: ComponentFixture<AppComponent>;
+  let loader: HarnessLoader;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [AppModule],
+      imports: [AppModule ,MatTabsModule],
     }).compileComponents();
 
     fixture = TestBed.createComponent(AppComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    loader = TestbedHarnessEnvironment.loader(fixture);
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should have the first tab selected', async () => {
+    const tabGroup = await loader.getHarness(MatTabGroupHarness);
+    const tabHarnesses = await tabGroup.getTabs()
+    const selectedTab = await tabHarnesses[0].isSelected();
+    expect(selectedTab).toBe(true);
+  });
+
+  it('should select the second tab when clicked', async () => {
+    const tabGroup = await loader.getHarness(MatTabGroupHarness);
+    const tabHarness = await tabGroup.getTabs();
+    const systolicInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='systolic']" }));
+    await systolicInputHarness.setValue('110');
+    const diastolicInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='diastolic']" }));
+    await diastolicInputHarness.setValue('70');
+    const submitButtonHarness = await loader.getHarness(MatButtonHarness.with({ selector: "[id='submit']" }));
+    await submitButtonHarness.click();
+    component.getSelectedIndex(1)
+    const selectedTab = await tabHarness[1].isSelected();
+    expect(selectedTab).toBe(true);
```

Comment:
did you check if this test fails when you break code (e.g, by not emitting the event)?